sklearn/preprocessing/_target_encoder.py

```diff
@@ -322,35 +381,142 @@ def _fit_encodings_all(self, X, y):
             dtype=np.int64,
             count=len(self.categories_),
         )
+        if self.target_type_ == "multiclass":
+            encodings = self._learn_multiclass_encodings(
+                X_ordinal,
+                y,
+                n_categories,
+                self.target_mean_,
+            )
+        else:
+            encodings = self._learn_binary_or_continuous_encoding(
+                X_ordinal,
+                y,
+                n_categories,
+                self.target_mean_,
+            )
+        self.encodings_ = encodings
+
+        return X_ordinal, X_known_mask, y, n_categories
+
+    def _learn_binary_or_continuous_encoding(
+        self, X_ordinal, y, n_categories, target_mean
+    ):
+        """Learn target encodings."""
         if self.smooth == "auto":
             y_variance = np.var(y)
-            self.encodings_ = _fit_encoding_fast_auto_smooth(
-                X_ordinal, y, n_categories, self.target_mean_, y_variance
+            encodings = _fit_encoding_fast_auto_smooth(
+                X_ordinal,
+                y,
+                n_categories,
+                target_mean,
+                y_variance,
             )
         else:
-            self.encodings_ = _fit_encoding_fast(
-                X_ordinal, y, n_categories, self.smooth, self.target_mean_
+            encodings = _fit_encoding_fast(
+                X_ordinal,
+                y,
+                n_categories,
+                self.smooth,
+                target_mean,
             )
+        return encodings
 
-        return X_ordinal, X_known_mask, y, n_categories
+    def _learn_multiclass_encodings(self, X_ordinal, y, n_categories, target_mean):
+        """Learn multiclass encodings.
+
+        Learn encodings for each class (c) then reorder encodings such that
+        the same features (f) are grouped together. `reorder_index` enables
+        converting from:
```

Comment:
Can this example be for 2 features and 3 classes? (By default, 2 classes will not use `_learn_multiclass_encodings`)

Reply:
good pickup!